File: code/Share2Calendar/app/src/main/java/com/xixilala/share2calendar/MainActivity.java
package com.xixilala.share2calendar;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView var2 = new TextView((Context)this);
        var2.setText(R.string.app_name);
        this.setContentView(var2);
    }
}

File: code/eventDispatch/parallel/src/main/java/com/xixilala/parallel/MainActivity.kt
package com.xixilala.parallel

import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AppCompatActivity
import com.xixilala.parallel.ui.main.SectionsPagerAdapter
import com.xixilala.parallel.ui.main.parallel.MyContainerViewGroup

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mt = findViewById<MyContainerViewGroup>(R.id.mt)
        val ids = arrayOf(R.layout.layout0,R.layout.layout1,R.layout.layout2).toIntArray()
        mt.setIds(ids)


    }
}